include/ht_file.h

```diff
@@ -19,8 +19,6 @@
 #include <ht_file_interface.h> //IFile
 #include <string> //std::string typedef
 #include <cstddef> //size_t typedef
```

Comment:
You use the typedef ```std::fstream``` here, and we don't want to have to rely on other headers for dependencies.  Better to just include it and have the include guard do its magic than have a change to ht_file_interface (or something) not allow this header to operate.